### PR TITLE
fix: add pod labels to service instances

### LINKED ISSUE #181: Service instances doesn't contain the labels from the endpoints backend pods
In Kubernetes service discovery strategy we should retrieve and add the labels from the backend pods for a given service, they could be needed during a custom load balancing strategy. Currently, only the service labels are being added to the service instances.

### PR DESCRIPTION
**Pull request by aureamunoz**
Fixes #181

### CODE REVIEW

**Review by cescoffier — APPROVED** [review]

File: service-discovery/kubernetes/src/main/java/io/smallrye/stork/servicediscovery/kubernetes/KubernetesServiceDiscovery.java [hunk 1]
```diff
@@ -66,25 +69,44 @@ public KubernetesServiceDiscovery(String serviceName, ServiceDiscoveryConfig con
 
     @Override
     public Uni<List<ServiceInstance>> fetchNewServiceInstances(List<ServiceInstance> previousInstances) {
-        Uni<List<Endpoints>> endpointsUni = Uni.createFrom().emitter(
+        Uni<Map<Endpoints, List<Pod>>> endpointsUni = Uni.createFrom().emitter(
                 emitter -> {
                     vertx.executeBlocking(future -> {
-                        List<Endpoints> endpoints = new ArrayList<>();
+                        Map<Endpoints, List<Pod>> items = new HashMap<>();
+                        List<Pod> backendPods = new ArrayList<>();
```
Comment (michalszynkiewicz) [inline service-discovery/kubernetes/src/main/java/io/smallrye/stork/servicediscovery/kubernetes/KubernetesServiceDiscovery.java:76]: it seems to me we're collecting pods for an  endpoint in this collection, shouldn't it be initialized in the for loop in line 81?

**Review by michalszynkiewicz — CHANGES_REQUESTED** [review]
I'm afraid the pod collection is shared between services. Isn't it?

File: service-discovery/kubernetes/src/main/java/io/smallrye/stork/servicediscovery/kubernetes/KubernetesServiceDiscovery.java (hunk 1 above)
Comment (aureamunoz) [reply service-discovery/kubernetes/src/main/java/io/smallrye/stork/servicediscovery/kubernetes/KubernetesServiceDiscovery.java]: Yes, good catch. I refactored to have it in the last commit

**Review by michalszynkiewicz — APPROVED** [review]

File: service-discovery/kubernetes/src/main/java/io/smallrye/stork/servicediscovery/kubernetes/KubernetesServiceDiscovery.java [hunk 2]
```diff
@@ -116,18 +143,24 @@ public KubernetesServiceDiscovery(String serviceName, ServiceDiscoveryConfig con
...
                     } else {
-                        Map<String, String> labels = endPoints.getMetadata().getLabels() != null
+                        Map<String, String> labels = new HashMap<>(endPoints.getMetadata().getLabels() != null
                                 ? endPoints.getMetadata().getLabels()
-                                : Collections.emptyMap();
+                                : Collections.emptyMap());
+                        Map<String, String> podLabels = pods.stream().filter(pod -> pod.getMetadata().getName().equals(podName))
+                                .findFirst().get().getMetadata().getLabels();
+                        for (Map.Entry<String, String> label : podLabels.entrySet()) {
+                            labels.putIfAbsent(label.getKey(), label.getValue());
```
[the remark this answers is not in the document]
Comment (aureamunoz) [reply service-discovery/kubernetes/src/main/java/io/smallrye/stork/servicediscovery/kubernetes/KubernetesServiceDiscovery.java:152]: Only one pod can have a given name at a time.

**Pull request merged**